contribute/style-guides/e2e.md

```diff
@@ -0,0 +1,140 @@
+# End to End test framework
+As our company grows our need for end to end (called e2e from now on) tests have also grown. Today we're using a minimal home grown solution built on top of Cypress for our e2e tests. 
+
+## Basic concepts
+A good introduction to some good e2e practices can be found [here](https://martinfowler.com/bliki/PageObject.html).
+- `Selector`, an unique identifier that is used from the e2e framework to retrieve an element from the Browser 
+- `Page`, an abstraction for an object that contains one or more `Selectors`
+- `Flow`, an abstraction that contains a sequence of actions on one or more `Pages` that can be reused and shared between tests
+
+## Basic example
+Lets start with a simple example with a single selector and for simplicity all examples will be in JSX.
+
+In our example app we have an input that we during our e2e test want to type some text in.
+```jsx harmony
+<div>
+    <input type="text" className="gf-form-input login-form-input"/>
+</div>
+```
+
+We could define a selector using `JQuery` [type selectors](https://api.jquery.com/category/selectors/) with a string like `'.gf-form-input.login-form-input'` but that would be brittle as style changes occur frequently and there is nothing that signalizes to future developers that this input is part of an e2e test.
```

Comment:
```suggestion
We could define a selector using `JQuery` [type selectors](https://api.jquery.com/category/selectors/) with a string like `'.gf-form-input.login-form-input'`, but that would be brittle as style changes occur frequently, and there is nothing that signals to future developers that this input is part of an e2e test.
```

Reply:
This sentence is awfully long. I suggest breaking it into several shorter sentences. It will be easier to read and understand.

Try to keep sentences under 25 words.

Reply:
Good feedback! I've broken it up into 2 sentences.